File: Second_approach/logger.js
// 🔹 Function to log successful requests
function logRequest(ip, method, path, status) {

    // Print log in consistent format
    console.log(`[LOG] ${ip} ${method} ${path} ${status}`);
}


// 🔹 Function to log blocked requests
function logBlocked(ip, method, path, reason) {

    // Print blocked log with reason
    console.log(`[BLOCKED] ${ip} ${method} ${path} ${reason}`);
}


// Export both functions so proxy.js can use them
module.exports = {
    logRequest,
    logBlocked
};

File: Go/internal/middleware/logger.go
package middleware

import (
	"fmt"
	"time"
)

func LogRequest(ip, path, method string) {
	fmt.Printf("[REQUEST] %s | %s %s | %s\n",
		time.Now().Format(time.RFC3339),
		method,
		path,
		ip,
	)
}

func LogBlocked(ip, reason string) {
	fmt.Printf("[BLOCKED] %s | IP: %s | Reason: %s\n",
		time.Now().Format(time.RFC3339),
		ip,
		reason,
	)
}